src/Nox.Types/Types/CurrencyNumber/CurrencyNumber.cs

```diff
@@ -1,12 +1,79 @@
+using System;
+using System.Collections.Generic;
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="CurrencyNumber"/> type and value object.
+/// Supports ETL Process's and allows to get the proper <see cref="CurrencyCode"/>
+/// </summary>    
+[Serializable]
+public sealed class CurrencyNumber : ValueObject<short, CurrencyNumber>
+{
+    private static HashSet<short> _currecyNumbers = new HashSet<short>()
```

Comment:
We have 154 Currency Numbers and 163 Currency Codes.
I think for consistency, we need to align the enum with the Wiki page (List of ISO 4217 currency codes)
https://en.wikipedia.org/wiki/ISO_4217

![image](https://github.com/NoxOrg/Nox.Generator/assets/131167535/4b6c59bd-e44c-4a5e-8a97-c62d7dc2853e)

Reply:
Once you finish this change of the Enum promote this Draft to a PR, please.